cpp/src/arrow/compute/api_vector.h

```diff
@@ -120,20 +120,29 @@ class ARROW_EXPORT SortOptions : public FunctionOptions {
   std::vector<SortKey> sort_keys;
 };
 
-// \brief Selection algorithm. Default is `NonStableSelect` which uses a Heap based
-// algorithm.
-enum class SelectKAlgorithm { NonStableSelect, StableSelect };
-
 /// \brief SelectK options
 class ARROW_EXPORT SelectKOptions : public FunctionOptions {
  public:
-  explicit SelectKOptions(int64_t k = -1, std::vector<SortKey> sort_keys = {},
-                          SelectKAlgorithm kind = SelectKAlgorithm::NonStableSelect);
+  explicit SelectKOptions(int64_t k = -1, std::vector<SortKey> sort_keys = {});
 
   constexpr static char const kTypeName[] = "SelectKOptions";
 
-  static SelectKOptions Defaults() {
-    return SelectKOptions{-1, {}, SelectKAlgorithm::NonStableSelect};
+  static SelectKOptions Defaults() { return SelectKOptions{-1, {}}; }
+
+  static SelectKOptions TopKDefault(int64_t k, std::vector<std::string> key_names = {}) {
+    std::vector<SortKey> keys;
+    for (const auto& name : key_names)
+      keys.emplace_back(SortKey(name, SortOrder::Descending));
```

Comment:
nit: please add braces for this sort of thing

Reply:
ok thanks